forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationService.java

```diff
@@ -251,6 +258,27 @@ public void verifyTPPRegistrationRequestAgainstSSA(RegistrationRequest registrat
 
     public Tpp registerTpp(ApiClientIdentity clientIdentity, RegistrationRequest oidcRegistrationRequest)
             throws DynamicClientRegistrationException {
+        
+        Optional<DirectorySoftwareStatement> jti = tppStoreService.findByOrganisationId(clientIdentity.getUsername())
+            .stream()
+            .map(Tpp::getSsa)
+            .map(ssa -> {
+                try {
+                    return objectMapper.readValue(ssa, DirectorySoftwareStatement.class);
+                } catch (IOException e) {
+                    log.error("registerTpp() SSA cannot be converted to a DirectorySoftwareStatement", e);
+                }
+            })
+            .filter(dss -> dss.getJti().equals(oidcRegistrationRequest.getJti()))
+            .findAny();
+        
+        if (jti.isPresent()) {
+            log.info("registerTpp() this tpp has already registered with an identical SSA. " +
+              "You can only onboard with a single software statement if you generate a new SSA");
+            return null;
```

Comment:
Maybe if registration has already happend with this SSA we could return the RegistrationResponse object that was originally returned for that registration. With a 200 response rather than a 201??

Reply:
Maybe we could do the same if we find the Software Statement has already been onboarded?? Check by software id?